ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -61,15 +74,24 @@ __yk_llvmwrap_symbolizer_find_code_sym(LLVMSymbolizer *Symbolizer,
 }
 
 // Load an LLVM module from an address.
-std::unique_ptr<Module> load_module(LLVMContext &Context, char *Ptr,
-                                    size_t Len) {
+//
+// This must only be called from getSrcMod() for correct synchronisation.
+void loadModule(char *Ptr, size_t Len) {
   auto Sf = StringRef(Ptr, Len);
   auto Mb = MemoryBufferRef(Sf, "");
   SMDiagnostic Error;
-  auto M = parseIR(Mb, Error, Context);
+  SharedCtx = std::make_unique<LLVMContext>();
+  auto M = parseIR(Mb, Error, *SharedCtx.getContext());
   if (!M)
     errx(EXIT_FAILURE, "Can't load module.");
-  return M;
+  SharedMod = ThreadSafeModule(std::move(M), SharedCtx);
+}
+
+// Load and return the in-binary LLVM module (or return the cached module if we
+// have already loaded it).
```

Comment:
Possibly dumb question: is caching this module safe or should we create a new one each time?

Reply:
Ah, so it's not "an LLVM module in a binary" but "a binary version of the LLVM module"? Should we not use different terminology here (I don't know what the right thing is: bitcode or ... ?)?

Reply:
It's the bitcode we load from the `.llvmbc` section of the interpreter binary.

So far @ptersilie and I have been using the term `SrcMod` and `DstMod` for the module we load from `.llvmbc` and the module we build and compile for the trace.

Reply:
Aha so we need to bikeshed the names :) "src" and "dst" suggest copying, but we're doing more than that here. How about "AOTMod" and "JITMod"?

Reply:
> "AOTMod" and "JITMod"

I like it!

Reply:
Works for me.